pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -134,18 +136,32 @@ public synchronized void shutDown() {
   public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService,
       @Nullable StreamObserver<Server.ServerResponse> responseObserver) {
     if (!queryRequest.isEnableTrace()) {
-      return processQueryInternal(queryRequest, executorService, responseObserver);
+      return (DataTable) processQueryInternal(queryRequest, executorService, false, responseObserver);
     }
     try {
       Tracing.getTracer().register(queryRequest.getRequestId());
-      return processQueryInternal(queryRequest, executorService, responseObserver);
+      return (DataTable) processQueryInternal(queryRequest, executorService, false, responseObserver);
     } finally {
       Tracing.getTracer().unregister();
     }
   }
 
-  private DataTable processQueryInternal(ServerQueryRequest queryRequest, ExecutorService executorService,
+  @Override
+  public BaseResultsBlock processQueryV2(ServerQueryRequest queryRequest, ExecutorService executorService,
       @Nullable StreamObserver<Server.ServerResponse> responseObserver) {
+    if (!queryRequest.isEnableTrace()) {
+      return (BaseResultsBlock) processQueryInternal(queryRequest, executorService, true, responseObserver);
+    }
+    try {
+      Tracing.getTracer().register(queryRequest.getRequestId());
+      return (BaseResultsBlock) processQueryInternal(queryRequest, executorService, true, responseObserver);
+    } finally {
+      Tracing.getTracer().unregister();
+    }
+  }
+
```

Comment:
This class needs a large refactoring if we want to return `BaseResultsBlock` for multi-stage. I created `DataTableBridge` to minimize the scope for this PR and plan to raise the refactoring PR right after this one. But I can take up the refactoring in this PR itself (this PR may then become large. Lmk if that's okay)